src/edu/stanford/nlp/ie/crf/CRFLogConditionalObjectiveFunction.java

```diff
@@ -776,39 +789,11 @@ public int domainDimension() {
    *
    * @return a 2D weight array
    */
-  public static double[][] to2D(double[] weights, List<Index<CRFLabel>> labelIndices, int[] map) {
-    double[][] newWeights = new double[map.length][];
-    int index = 0;
-    for (int i = 0; i < map.length; i++) {
-      int labelSize = labelIndices.get(map[i]).size();
-      newWeights[i] = new double[labelSize];
-      try {
-        System.arraycopy(weights, index, newWeights[i], 0, labelSize);
-      } catch (Exception ex) {
-        log.info("weights: " + Arrays.toString(weights));
-        log.info("newWeights["+i+"]: " + Arrays.toString(newWeights[i]));
-        throw new RuntimeException(ex);
-      }
-      index += labelSize;
-    }
-    return newWeights;
-  }
-
-  public double[][] to2D(double[] weights) {
```

Comment:
This is actually used by a subclass of CRFLogConditionalObjectiveFunction which we have hidden in our internal codebase.  Something we didn't want to publicly release because of licensing issues around some of the other libraries it uses.  

Any particular reason for removing it?  Just general cleanliness?

Actually we could easily move it to the subclass I think, so not really a problem

Reply:
I realized the other class was actually using this as a library, so I left the original here (with similar changes to the ones you made).